backend/pkg/api/data_access/vdb_rewards.go

```diff
@@ -740,206 +720,198 @@ func (d *DataAccessService) GetValidatorDashboardRewardsChart(ctx context.Contex
 			elDs = elDs.
 				SelectAppend(goqu.L("?::smallint AS result_group_id", t.DefaultGroupId))
 		} else {
-			rewardsDs = rewardsDs.
-				SelectAppend(goqu.L("v.group_id AS result_group_id")).
-				GroupByAppend(goqu.L("result_group_id")).
-				OrderAppend(goqu.L("result_group_id").Asc())
 			elDs = elDs.
 				SelectAppend(goqu.L("v.group_id AS result_group_id")).
 				GroupByAppend(goqu.L("result_group_id")).
 				OrderAppend(goqu.L("result_group_id").Asc())
 		}
 	} else {
 		// In case a list of validators is provided set the group to the default id
-		rewardsDs = rewardsDs.
-			SelectAppend(goqu.L("?::smallint AS result_group_id", t.DefaultGroupId)).
-			Where(goqu.L("e.validator_index IN ?", dashboardId.Validators))
 		elDs = elDs.
 			SelectAppend(goqu.L("?::smallint AS result_group_id", t.DefaultGroupId)).
 			Where(goqu.L("b.proposer = ANY(?)", pq.Array(dashboardId.Validators)))
 	}
+	elDs = elDs.
+		With("epoch_ranges(epoch_start, epoch_end)", goqu.L("(SELECT * FROM unnest(?::int[], ?::int[]))", pq.Array(epochStarts), pq.Array(epochEnds))).
+		InnerJoin(goqu.L("epoch_ranges"), goqu.On(goqu.L("b.epoch BETWEEN epoch_ranges.epoch_start AND epoch_ranges.epoch_end"))).
+		Where(goqu.L("b.epoch BETWEEN ? AND ?", epochStarts[0], epochEnds[len(epochEnds)-1]))
+	return elDs
+}
+
+func (d *DataAccessService) GetValidatorDashboardRewardsChart(ctx context.Context, dashboardId t.VDBId, groupIds []int64, protocolModes t.VDBProtocolModes, aggregation enums.ChartAggregation, afterTs uint64, beforeTs uint64) (*t.ChartData[int, decimal.Decimal], error) {
+	// @DATA-ACCESS incorporate protocolModes
+	// bar chart for the CL and EL rewards for each group for each epoch.
+	// NO series for all groups combined except if AggregateGroups is true.
+	// series id is group id, series property is 'cl' or 'el'
+	ret := &t.ChartData[int, decimal.Decimal]{}
+
+	if len(groupIds) == 0 {
+		return ret, nil
+	}
 
+	requestedAllGroups := dashboardId.AggregateGroups || slices.Contains(groupIds, t.AllGroups)
 	// ------------------------------------------------------------------------------------------------------------------
-	// Build the main query and get the data
-	queryResult := []struct {
+	// CL
+
+	clDs, err := buildRewardChartClDs(dashboardId, groupIds, afterTs, beforeTs, aggregation, requestedAllGroups)
+	if err != nil {
+		return nil, fmt.Errorf("error building CL rewards query: %w", err)
+	}
+	type clResult struct {
 		Timestamp  time.Time `db:"timestamp"`
 		EpochStart uint64    `db:"epoch_start"`
 		EpochEnd   uint64    `db:"epoch_end"`
 		GroupId    uint64    `db:"result_group_id"`
 		ClRewards  int64     `db:"cl_rewards"`
-	}{}
-
-	query, args, err := rewardsDs.Prepared(true).ToSQL()
-	if err != nil {
-		return nil, fmt.Errorf("error preparing query: %w", err)
 	}
 
-	err = d.clickhouseReader.SelectContext(ctx, &queryResult, query, args...)
+	queryResult, err := runQueryRows[[]clResult](ctx, d.clickhouseReader, clDs)
```

Comment:
Pls do analog to `elQueryResult`:
```suggestion
	clQueryResult, err := runQueryRows[[]clResult](ctx, d.clickhouseReader, clDs)
```

Reply:
Fixed, ptal